base/hamt.jl

```diff
@@ -75,26 +75,23 @@ function HAMT{K,V}(k::K, v) where {K, V}
 end
 HAMT(k::K, v::V) where {K, V} = HAMT{K,V}(K, V)
 
-hash(x) = objectid(x)::UInt
-hash(x, h::UInt) = hash(x) - 3h
-
 struct HashState{K}
     key::K
     hash::UInt
     depth::Int
     shift::Int
 end
-HashState(key)= HashState(key, hash(key), 0, 0)
+HashState(key)= HashState(key, objectid(key), 0, 0)
 # Reconstruct
-HashState(key, depth, shift) = HashState(key, hash(key, UInt(depth ÷ BITS_PER_LEVEL)), depth, shift)
+HashState(key, depth, shift) = HashState(key, hash(objectid(key), UInt(depth ÷ BITS_PER_LEVEL)), depth, shift)
```

Comment:
@vtjnash this reminded me of a question I had during the original development.

Is there a better way of doing: `hash(objectid(key), UInt(depth ÷ BITS_PER_LEVEL))`?

Reply:
Doesn't seem like it, unless you wanted to call it something more specific like `hashmix(objectid(key), UInt(depth + BITS_PER_LEVEL` (since libsupport calls this bitmix)